### PR TITLE
moved math stuff from common to numbers

### PR DESCRIPTION
**Pull request by craiglowe**
Moving math from common to numbers

### CODE REVIEW

File: numbers/minMax.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+func MinFloat64(a float64, b float64) float64 {
+	if a <= b {
+		return a
+	} else {
+		return b
+	}
+}
+
```
Comment (rimangan) [inline numbers/minMax.go:82]: Are these triple functions commonly used? Would it be better to just have a function for a slice of ints? Not sure what is best stylistically moving forward

**Review by rimangan — APPROVED** [review]
Looks great! This is an important fix!

File: numbers/minMax.go (hunk 1 above)
Comment (edotau) [reply numbers/minMax.go:82]: I am against using a slice for this function. You would have to start looping through the slice and you have to allocate additional memory to keep track of the the Min/Max. Using individual numbers also ensures you allocate memory on the stack (at least that is my numberstanding).

**Review by edotau — APPROVED** [review]

File: numbers/minMax.go (hunk 1 above)
Comment (edotau) [reply numbers/minMax.go:82]: No, i am against using slices for this. You also have to program a loop and also have to allocate additional memory to keep track of the current max/min, whereas here no additonal memory is needed to return an answer. When you work with simple data types (like ints and strings) memory allocation is more likely on the stack. When you start to use more slices, maps or complex structs, if you are sloppy or not careful, the memory escapes to the heap (at least that is my understanding).

Comment (craiglowe) [reply numbers/minMax.go:82]: I think it makes sense to keep these functions for the reasons Eric gave.  However, in the future we could add versions of min/max that operate on a slice if that would be useful for people.

Comment (edotau) [reply numbers/minMax.go:82]: The slice version of min/max essentially would look the same as a sort, either ascending or descending order, returning either the head or tail, is that right?

Comment (DanielSnellings) [reply numbers/minMax.go:82]: Not necessarily. A min/max would require far fewer comparisons than a true sort.

**Review by DanielSnellings — APPROVED** [review]

**Pull request merged**